plugin/src/main/java/io/snyk/eclipse/plugin/runner/ProcessRunner.java

```diff
@@ -120,9 +120,15 @@ private void setupProcessBuilderBase(ProcessBuilder pb) {
       }
     }
 
+    String authMethod = Preferences.getInstance().getPref(Preferences.AUTHENTICATION_METHOD);
     String token = Preferences.getInstance().getAuthToken();
-    if (token != null)
-      pb.environment().put(EnvironmentConstants.ENV_SNYK_TOKEN, Preferences.getInstance().getAuthToken());
+    if (token != null && authMethod.equals(Preferences.AUTH_METHOD_TOKEN)) {
```

Comment:
we can avoid a possible NPE by inverting the equals to `Preferences.AUTH_METHOD_TOKEN.equals(...)`

Reply:
But maybe then the if should check against `AUTH_METHOD_OAUTH` (inverting it) so that the else always assumes token authentication. This will be more resilient for backwards compatibility and upgrades.

Reply:
updated